src/notifications/ldn.ts

```diff
@@ -0,0 +1,84 @@
+/**
+ * Copyright 2020 Inrupt Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to use,
+ * copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
+ * Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+import { LitDataset, Iri, Thing, IriString } from "../interfaces";
+import { dataset, DataFactory } from "../rdfjs";
+import { fetch } from "../fetcher";
+
+/**
+ * Perform partial inbox discovery (https://www.w3.org/TR/ldn/#discovery) by only checking
+ * resource content. If the inbox is only advsertised in the resource metadata, it won't be
```

Comment:
**Edit:** Ah, there's `fetchInbox` below. Is there a reason someone would want to use this function rather than that one?

---

We _could_ support discovery from Link headers as well? (Might be a good moment to decide attaching the whole parsed link header to `resourceInfo`, possibly tagged as `internal_`, as an escape hatch.)

Also:

```suggestion
 * resource content. If the inbox is only advertised in the resource metadata, it won't be
```

Reply:
Basically there exists two functions to cover two use case (and allow optimizations): either you want to send a notification to a target resource for which you know the IRI, then you `fetchInbox`, or you have a resource fetched already and want to look up its inbox (but not perform one, potentially two additional HTTP requests), and then you `discoverInbox`. At least that's the initial intent :)